app/models/notification.rb

```diff
@@ -0,0 +1,22 @@
+class Notification < ApplicationRecord
+  include PgSearch::Model
+  belongs_to :recipient, class_name: 'User'
+  belongs_to :actor, class_name: 'User', optional: true
+  belongs_to :notifiable, polymorphic: true, optional: true
+
+  enum :event => { topic_created: "topic.created", post_created: "post.created" }
+
+  scope :unread, -> { where(read_at: nil) }
+  scope :read, -> { where.not(read_at: nil) }
+
+  NOTIFICATION_READ = :read
+  NOTIFICATION_UNREAD = :unread
+
+  VALID_STATUS_TYPES = [
+    NOTIFICATION_READ,
+    NOTIFICATION_UNREAD
+  ].freeze
```

Comment:
I don't think these values should be _kept_ in here. I understand that this is being used as a common holding space for values used elsewhere, but this goes against the usual pattern of these being possible values of a `string` type column on the table.